runtime/src/assets.rs

```diff
@@ -83,6 +150,11 @@ impl TryFromSymbol for CurrencyId {
             id if id == KSM.symbol() => Ok(Token(KSM)),
             id if id == KBTC.symbol() => Ok(Token(KBTC)),
             id if id == KINT.symbol() => Ok(Token(KINT)),
+            // Does the first character match the lend token prefix?
+            id if id.chars().next() == Some(LEND_TOKEN_SYMBOL_PREFIX) => {
```

Comment:
The unfortunate problem with this is that parsing would fail if we were to register any other asset which begins with `Q`, we should somehow always fall-through to `get_foreign_asset_by_symbol`.

Reply:
Found a pretty neat solution using an if-let match guard. This feature is experimental however - no bugs have been reported but it's just not well-documented enough: https://github.com/rust-lang/rust/issues/51114 (the issue thread was active within the last month).